### DISCUSSION TITLE
intended purpose of server/api

### DISCUSSION THREAD

**Discussion by isimmons** [nuxt/nuxt#16434] [Questions] [answered]
I've watched a couple of courses/tutorials where server/api is treated as a public api for my app to receive/respond to incoming api requests from other external apps. This is normal to me.

I've also seen a couple of tutorials that were accessing other apis (notion and coinlore) from a page in our project (lets call it myapp). Instead of writing a function, composable, or whatever and making a useFetch call to the external api, they make a $fetch call to myapp/api/foo and then myapp/api/foo makes a second api call to the actual external api and then returns the response back to the myapp/page that called it. This works but seems very strange to me. 

It seems like me calling myself on the phone to remind myself to call someone else instead of just calling them in the first place?

So I'm wondering what exactly is the intended use of the api and are there situations where the above example makes sense?

**Comment by TheAlexLichter** [comment, marked as answer]
Use cases can be different and there is no "you should use it for X". You can write your whole logic in it, use it for "proxying" your API calls and also how you mentioned.

The pattern you've described is becoming more and more common. Check out "Backend for Frontend". It makes a lot of sense so you can:

* have a uniform interface when bundling multiple API calls
* add e.g. caching and storage layers in between to improve speed and fault tolerance (e.g. when 3rd party APIs are not reachable)

**Comment by isimmons** [comment]
Thanks @manniL I googled "Backend for Frontend" Looks like I have a lot to read on the subject.
